responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -311,6 +313,14 @@ def _validate_model_analysis_input_parameters(
                             'The model passed cannot be used for'
                             ' getting predictions via predict_proba()'
                         )
+                small_train_features_after = list(small_train_data.columns)
+
+                if small_train_features_before != small_train_features_after:
```

Comment:
should we also have this check after predict_proba() test as well (just below). It could be that predict() doesn't change the input data but predict_proba() changes the input data.

Reply:
oh the check is after both the predict and predict_proba

Reply:
I suppose I could add a separate check for predict_proba so as to have a custom error message though

Reply:
done, added separate checks both after predict and predict_proba methods, and added another test to validate each case